apps/browser/qml/pages/components/TabGridView.qml

```diff
@@ -41,6 +41,24 @@ SilicaGridView {
     signal closeAllCanceled
     signal closeAllPending
 
+    function closeAllTabs() {
+        remorsePopup = Remorse.popupAction(
+                    tabGridView,
+                    //% "Closed all tabs"
+                    qsTrId("sailfish_browser-closed-all-tabs"),
+                    function() {
+                        tabGridView.closeAll()
+                        remorsePopup = null
+                    })
+        closingAllTabs = true
```

Comment:
The remorse timer creates some complexity, because it's possible to switch between normal and private browsing while the remorse is still running. This can mean that the user selects "Close all tabs" in private mode, but it's the normal browsing tabs which get closed. These are the steps I followed:
1. Open tab view.
2. Select batman to show the private tabs.
3. Select "Close all tabs" from the menu.
4. While the remorse timer is still running, open the menu again and select "New tab".

In this case, the model switches to normal browsing mid-remorse and the wrong set of tabs gets closed. I'm not sure what the solution is, but one way might be to block the UI from switching views while the remorse is running.

Reply:
done

Reply:
This is definitely much improved, but I'm still experiencing some unexpected behaviour here, with the tab view getting confused about whether there are, or are not, tabs open. Here's the sequence I follow to hit the unexpected behaviour:

1. Open browser and clear all tabs from both normal and private modes.
2. Close and re-open browser.
3. Wait for the default homepage to load.
4. Open the tab view.
5. Select the "private mode" icon at the top of the screen.
6. Select the "normal mode" icon at the top of the screen.
7. Press on the "x" for the single open tab to close it.

At this point the tab count goes to zero and the "Push button plus to create a new tab" text appears in the background, but the tab isn't removed. It also seems to be possible to get the "private" tab icon confused this way (see the screenshot). I'm not sure exactly why this is happening I'm afraid.

![confused-tabs](https://user-images.githubusercontent.com/1446122/123796496-da861080-d8ed-11eb-97b6-db35a0c80e40.png)

Reply:
Easier to reproduce
1) Launch browser
2) Wait for the home page to load
3) Switch to private mode
4) Go to normal mode
5) Remove tab

Not perceived 
```
        beginRemoveRows(QModelIndex(), index, index);
        m_tabs.removeAt(index);
        endRemoveRows();
```

Reply:
I got around this, but this is a terrible solution (
    ```
    Connections {
        target: model
        // Force update GridView when deleting a tab
        onTabClosed: {
            var oldModel = tabGridView.model
            tabGridView.model = undefined
            tabGridView.model = oldModel
        }
    }
    ```

Reply:
I don't see the underlying reason for this I'm afraid, but I can confirm that your solution works here too.

Reply:
How is it possible to catch such an error, according to the previous instructions, I cannot

Reply:
Just to clarify: the error no longer happens after your fix (at least, I'm not able to find a way to make it happen).